tutorials/polkadot-sdk/parachains/build-custom-pallet/build-pallet.md

```diff
@@ -0,0 +1,290 @@
+
+Below are the events defined for this pallet:
+
+- **CounterValueSet** - is emitted when the counter is explicitly set to a new value. This event includes the counter's updated value
+
+- **CounterIncremented** - is emitted after a successful increment operation. It includes:
+
+    - The new counter value
+    - The account responsible for the increment
+    - The amount by which the counter was incremented
+
+- **CounterDecremented** - is emitted after a successful decrement operation. It includes:
+
+    - The new counter value
+    - The account responsible for the decrement
+    - The amount by which the counter was decremented
+
+Define the events in the pallet as follows:
+
+```rust
+--8<-- 'code/tutorials/polkadot-sdk/parachains/build-custom-pallet/build-pallet/lib.rs:25:51'
+```
+
+### Add Storage Items
+
+Storage items are used to manage the pallet's state. This pallet defines two items to handle the counter's state and user interactions:
+
+- **CounterValue** - a single storage value that keeps track of the current value of the counter. This value is the core state variable manipulated by the pallet's functions
+
+- **UserInteractions** - a storage map that tracks the number of times each account interacts with the counter
+  
+Define the storage items as follows:
+
+```rust
+--8<-- 'code/tutorials/polkadot-sdk/parachains/build-custom-pallet/build-pallet/lib.rs:53:59'
+```
+
+### Implement Custom Errors
+
+The `#[pallet::error]` macro defines a custom `Error` enum to handle specific failure conditions within the pallet. Errors help provide meaningful feedback to users and external systems when an extrinsic cannot be completed successfully. They are critical for maintaining the pallet's clarity and robustness.
+
+To add custom errors, use the `#[pallet::error]` macro to define the `Error` enum. Each variant represents a unique error that the pallet can emit, and these errors should align with the logic and constraints of the pallet. 
+
+Add the following errors to the pallet:
+
+```rust
+--8<-- 'code/tutorials/polkadot-sdk/parachains/build-custom-pallet/build-pallet/lib.rs:61:71'
+```
+
+### Implement Calls
+
+The `#[pallet::call]` macro defines the dispatchable functions (or calls) the pallet exposes. These functions allow users or the runtime to interact with the pallet's logic and state. Each call includes comprehensive validations, modifies the state, and optionally emits events to signal successful execution.
+
+The structure of the dispatchable calls in this pallet is as follows:
+
+```rust
+--8<-- 'code/tutorials/polkadot-sdk/parachains/build-custom-pallet/build-pallet/call_structure.rs'
+```
+
+Below you can find the implementations of each dispatchable call in this pallet:
+
+???+ function "set_counter_value(origin: OriginFor<T>, new_value: u32) -> DispatchResult"
```

Comment:
```suggestion
???- function "set_counter_value(origin: OriginFor<T>, new_value: u32) -> DispatchResult"
```

Because these are sorta long, lets not expand them and allow users to do so